### PR TITLE
feat: Add httpOnly cookie attribute support

### PR DESCRIPTION
**Pull request by maximehamou**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Added httpOnly attribute support for local and refresh providers + updated the docs accordingly

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by phoenix-ru — DISMISSED** [review]

**Comment by phoenix-ru** [comment]
@maximehamou Could you please fix lint issues? Aside from that, code lgtm

**Comment by maximehamou** [comment]
> @maximehamou Could you please fix lint issues?

No problem, I will shortly.

However, I think it’s a little bit trickier than expected regarding how the update works. After further tests, the cookie is not saved. If I set `httpOnlyCookieAttribute` to `false`, and I force `httpOnly` with the checkbox in storage section, I couldn't log out anymore...

File: src/module.ts [hunk 1]
```diff
@@ -92,15 +93,17 @@ const defaultsByBackend: {
       maxAgeInSeconds: 5 * 60, // 5 minutes
       sameSiteAttribute: 'none',
       secureCookieAttribute: false,
-      cookieDomain: ''
+      cookieDomain: '',
+      httpOnlyCookieAttribute: false,
```
Comment (zoey-kaiser) [inline src/module.ts:97]:
```suggestion
      httpOnlyCookieAttribute: false
```

File: src/module.ts [hunk 2]
```diff
@@ -92,15 +93,17 @@ const defaultsByBackend: {
...
     },
     refreshToken: {
       signInResponseRefreshTokenPointer: '/refreshToken',
       refreshRequestTokenPointer: '/refreshToken',
       cookieName: 'auth.refresh-token',
       maxAgeInSeconds: 60 * 60 * 24 * 7, // 7 days
       secureCookieAttribute: false,
-      cookieDomain: ''
+      cookieDomain: '',
+      httpOnlyCookieAttribute: false,
```
Comment (zoey-kaiser) [inline src/module.ts:106]:
```suggestion
      httpOnlyCookieAttribute: false
```

File: src/runtime/composables/local/useAuthState.ts [hunk 3]
```diff
@@ -30,7 +30,8 @@ export const useAuthState = (): UseAuthStateReturn => {
     domain: config.token.cookieDomain,
     maxAge: config.token.maxAgeInSeconds,
     sameSite: config.token.sameSiteAttribute,
-    secure: config.token.secureCookieAttribute
+    secure: config.token.secureCookieAttribute,
+    httpOnly: config.token.httpOnlyCookieAttribute,
```
Comment (zoey-kaiser) [inline src/runtime/composables/local/useAuthState.ts:34]:
```suggestion
    httpOnly: config.token.httpOnlyCookieAttribute
```

File: src/runtime/composables/refresh/useAuthState.ts [hunk 4]
```diff
@@ -20,7 +20,8 @@ export const useAuthState = (): UseAuthStateReturn => {
       domain: config.refreshToken.cookieDomain,
       maxAge: config.refreshToken.maxAgeInSeconds,
       sameSite: 'lax',
-      secure: config.refreshToken.secureCookieAttribute
+      secure: config.refreshToken.secureCookieAttribute,
+      httpOnly: config.refreshToken.httpOnlyCookieAttribute,
```
Comment (zoey-kaiser) [inline src/runtime/composables/refresh/useAuthState.ts:24]:
```suggestion
      httpOnly: config.refreshToken.httpOnlyCookieAttribute,
```

**Review by zoey-kaiser — APPROVED** [review]

**Pull request merged**